ras/smtstate.py

```diff
@@ -67,3 +73,26 @@ def test(self):
                     self.log.info("SMT load is successful for SMT=%s" % j)
                 else:
                     self.fail("smt load failed")
+
+    def test_smt_state_switching(self):
+        """
+        Test to check the time taken for switching between different
+        SMT levels. The time taken must not exceed beyond 5 minutes.
+        """
+        process.system("ppc64_cpu --smt=on")
+
+        for i in [0, 1, 2, 4]:
+            if i == 0:
+                cmd_output = process.run(
+                    "/usr/bin/time -p ppc64_cpu --smt off", shell=True)
+            elif i == 1:
+                cmd_output = process.run(
+                    "/usr/bin/time -p ppc64_cpu --smt on", shell=True)
+            else:
+                cmd_output = process.run(
+                    "/usr/bin/time -p ppc64_cpu --smt %s", i, shell=True)
+
+            if (int(cmd_output.duration) > self.time_in_minutes):
+                self.fail("ERROR : SMT has taken longer than expected")
```

Comment:
here also please FAIL: SMT has taken longer than expected

Reply:
Fine. Will change it!

Reply:
Done